README.md

```diff
@@ -1 +1,176 @@
+
+		}
+	}
+}
+
+angular
+	.module('app')
+	.directive('someDirective', SomeDirective);
+```
+
+Now it's important to note that `elem` isn't *actually* the raw DOM node. It is a jqLite (or jQuery if you have loaded jQuery before Angular) element. However, we can access the raw DOM node via `elem[0]`.
+
+Let's add a click event to our `<span />`.
+
+```js
+function SomeDirective() {
+	return {
+		template: [
+			'<div>',
+				'<span>Click on me!</span>',
+			'</div>'
+		].join(''),
+		link: function (scope, elem, attrs) {
+			var actualElement = elem[0];
+
+			var spanElement = actualElement.querySelector('span');
+
+			spanElement.addEventListener('click', function () {
+				alert('You clicked me!');
+			});
+		}
+	}
+}
+
+angular
+	.module('app')
+	.directive('someDirective', SomeDirective);
+```
+
+Here, we will get an alert showing when the user clicks on the span. Awesome! Now, say that we want to actually update the `scope` values when the user clicks on the span - how do we do this? First of all, let's put a scope value in our view and controller:
+
+```js
+function SomeDirective() {
+	return {
+		template: [
+			'<div>',
+				'<span>Click on me!</span>',
+				'{{ status }}',
+			'</div>'
+		].join(''),
+		controller: function ($scope) {
+			$scope.status = 'Not clicked!';
+		},
+		link: function (scope, elem, attrs) {
+			var actualElement = elem[0];
+
+			var spanElement = actualElement.querySelector('span');
+
+			spanElement.addEventListener('click', function () {
+				scope.status = 'Clicked!';
+			});
+		}
+	}
+}
+
+angular
+	.module('app')
+	.directive('someDirective', SomeDirective);
+```
+
+Now you can see that we're updating `scope` in our link function. However, this won't actually work. We're outside the scope of Angular here, so Angular isn't able to tell that we've updated our scope.
+
+However, we can manually push Angular to update, by using `scope.$apply()`. Don't worry about this magic, we'll go into detail with this soon.
+
+```js
+function SomeDirective() {
+	return {
+		template: [
+			'<div>',
+				'<span>Click on me!</span>',
+				'{{ status }}',
+			'</div>'
+		].join(''),
+		controller: function ($scope) {
+			$scope.status = 'Not clicked!';
+		},
+		link: function (scope, elem, attrs) {
+			var actualElement = elem[0];
+
+			var spanElement = actualElement.querySelector('span');
+
+			spanElement.addEventListener('click', function () {
+				scope.status = 'Clicked!';
+
+				scope.$apply();
+			});
+		}
+	}
+}
+
+angular
+	.module('app')
+	.directive('someDirective', SomeDirective);
+```
+
+Sorted! This will update our `scope` with the new value.
+
+## Update the controller
+
+One problem - we don't *really* use `$scope` anymore - we use controller values. Well, much like when we required the parent controller in our previous README, we can actually request our own directives controller for use in the link function.
+
+To do this, we add `require` with the value `ngController`.
+
+```js
```

Comment:
break this code sample into two parts - first, we're adding the require property and requiring `ngController`. Then, in part 2, we can use our trusty controllerAs syntax instead of scope.